daemon/src/oracle.rs

```diff
@@ -310,7 +310,7 @@ impl xtra::Actor for Actor {
         let this = ctx.address().expect("we are alive");
         self.tasks.add(
             this.clone()
-                .send_interval(std::time::Duration::from_secs(5), || Sync),
+                .send_interval(std::time::Duration::from_secs(30), || Sync),
```

Comment:
🤔 Should we have two intervals?

We know that we only get one attestation per hour, but it is very important to get attestations as soon as possible. We could optimise this further but I think it's acceptable to check for a new attestation every 30 seconds.

But the announcements are updated at a completely different rate. We look 25 hours in the future, but we could look months ahead (e.g. https://outcome.observer/h00.ooo/x/BitMEX/BXBT/2022-07-20T01:00:00.price?n=20).

I would recommend to look 48 hours ahead and only run `ensure_having_announcements` once per hour. Would leave a lot of space for further optimisation, but the gains would be small given that we would already be doing

https://github.com/itchysats/itchysats/blob/5e6d8e63c1636eb6d7de725b6c489622e3da466b/daemon/src/oracle.rs#L136-L141

48 times per hour, which is pretty inexpensive.

Reply:
I think it is acceptable to check every 30 seconds for now. We can reduce this to check even less often but I'm not sure how much gain we would get from this.

Reply:
Sure, we don't have to go to 60 minutes now.

But why 30 seconds for announcements and 60 seconds for attestations? It would make more sense the other way around for me.

Reply:
Fair point. what timing would you chose?

Reply:
Maybe:

- Attestations every 30 seconds.
- Announcements every minute.

Reply:
works for me :)